mesonbuild/compilers/mixins/clike.py

```diff
@@ -104,7 +104,7 @@ def to_native(self, copy: bool = False) -> T.List[str]:
                     bad_idx_list += [i, i + 1]
                 elif each.startswith('-isystem=') and self._cached_realpath(each[9:]) in real_default_dirs:
                     bad_idx_list += [i]
-                elif self._cached_realpath(each[8:]) in real_default_dirs:
+                elif len(each) > 8 and self._cached_realpath(each[8:]) in real_default_dirs:
```

Comment:
I would say this is the wrong solution, because up above we check if `each == '-isystem'` which is the exact condition you are asking for here.

But that condition shouldn't be dropping down to elif, when for example `self._cached_realpath(new[i + 1]) in real_default_dirs)` is False. Even if it didn't hit bug you are pointing to, it's a bunch of useless comparisons we don't need to perform...

Reply:
Agree. I just updated the fix to move `self._cached_realpath(new[i + 1]) in real_default_dirs)` into a next-level if-statement so that it being false does not cause the other `elif` for `-isystem=` and `-isystemXXX` to be checked.